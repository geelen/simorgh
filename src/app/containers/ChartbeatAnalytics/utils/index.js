import Cookie from 'js-cookie';
import onClient from '../../../lib/utilities/onClient';
import { getPromoHeadline } from '../../../lib/analyticsUtils/article';
import { getPageTitle } from '../../../lib/analyticsUtils/frontpage';
import { getReferrer } from '../../../lib/analyticsUtils';

const ID_COOKIE = 'ckns_sylphid';

export const chartbeatUID = 50924;
export const useCanonical = true;
export const chartbeatSource = '//static.chartbeat.com/js/chartbeat.js';

const buildSectionArr = (service, value, type) => [
  `${service} - ${value}`,
  ...(type ? [`${service} - ${value} - ${type}`] : []),
];

const buildServiceType = (service, type) => [`${service} - ${type}`];

const capitalize = s => s.charAt(0).toUpperCase() + s.slice(1);

export const getSylphidCookie = () =>
  onClient() ? Cookie.get(ID_COOKIE) : null;

export const getDomain = service => {
  const serviceLower = service.toLowerCase();

  return serviceLower === 'news' ? 'bbc.co.uk' : `${service}.bbc.co.uk`;
};

export const getType = (pageType, shorthand = false) => {
  switch (pageType) {
    case 'frontPage':
    case 'index':
      return shorthand ? 'IDX' : 'Index';
    case 'article':
      return shorthand ? 'ART' : 'New Article';
    default:
      return null;
  }
};

export const buildSections = (service, type, producer, chapter) => {
  const addProducer = producer && service !== producer;
  const serviceCap = capitalize(service);
  const pageType = getType(type, true);

  const parts = [
    serviceCap,
    ...(type ? buildServiceType(serviceCap, pageType) : []),
    ...(addProducer ? buildSectionArr(serviceCap, producer, pageType) : []),
    ...(chapter ? buildSectionArr(serviceCap, chapter, pageType) : []),
  ];

  return parts.join(', ');
};

export const getTitle = (pageType, pageData, brandName) => {
  switch (pageType) {
    case 'frontPage':
    case 'index':
      return getPageTitle(pageData, brandName);
    case 'article':
      return getPromoHeadline(pageData);
    default:
      return null;
  }
};

export const getConfig = ({
  isAmp,
  platform,
  pageType,
  data,
  brandName,
  env,
  service,
  origin,
  previousPath,
}) => {
  const referrer = getReferrer(platform, origin, previousPath);
  const title = getTitle(pageType, data, brandName);
  const domain = env !== 'live' ? getDomain('test') : getDomain(service);
  const sections = buildSections(service, pageType);
  const cookie = getSylphidCookie();
  const type = getType(pageType);
  const currentPath = onClient() && window.location.pathname;
  return {
    domain,
    sections,
    uid: chartbeatUID,
    title,
    virtualReferrer: referrer,
    ...(isAmp && { contentType: type }),
    ...(!isAmp && {
      type,
      useCanonical,
      path: currentPath,
    }),
    ...(cookie && { idSync: { bbc_hid: cookie } }),
  };
};
