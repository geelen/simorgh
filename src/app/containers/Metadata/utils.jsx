import React from 'react';

export const getIconAssetUrl = (service, size) =>
  `https://news.files.bbci.co.uk/include/articles/public/${service}/images/icons/icon-${size}.png`;

const createIconLinks = (service, iconSizes, iconType) => {
  return iconSizes.map(size => {
    const iconAssetUrl = getIconAssetUrl(service, size);
    const key = `${service}-${size}`;

    if (iconType === 'icon') {
      return (
        <link
          key={key}
          rel="icon"
          type="image/png"
          href={iconAssetUrl}
          sizes={size}
        />
      );
    }

    return (
      <link key={key} rel="apple-touch-icon" sizes={size} href={iconAssetUrl} />
    );
  });
};

export const getIconLinks = (service, iconSizes) => {
  if (!iconSizes) {
    return null;
  }
  const iconTypes = Object.keys(iconSizes);
  return iconTypes.map(iconType => {
    return createIconLinks(service, iconSizes[iconType], iconType);
  });
};

export const renderAmpHtml = (ampLink, isAmp) => {
  if (isAmp) {
    return null;
  }
  return <link rel="amphtml" href={ampLink} />;
};

export const getAppleTouchUrl = service => {
  const assetsPath = process.env.SIMORGH_PUBLIC_STATIC_ASSETS_PATH || '/';
  const separatorSlash = assetsPath[assetsPath.length - 1] !== '/' ? '/' : '';

  return [
    process.env.SIMORGH_PUBLIC_STATIC_ASSETS_ORIGIN,
    assetsPath,
    separatorSlash,
    service,
    '/images/icons/icon-192x192.png',
  ].join('');
};

export const renderAlternateLinks = link => (
  <link
    rel="alternate"
    href={link.href}
    hrefLang={link.hrefLang}
    key={link.hrefLang}
  />
);
