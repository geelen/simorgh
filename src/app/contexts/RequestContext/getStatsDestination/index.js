/* Returns stats destnation for ATI based on origin, service and env
   see table on this issue https://github.com/bbc/simorgh/issues/2995
*/
const getStatsDestination = ({ isUK = true, env = 'test', service }) => {
  let destination = '';
  switch (service) {
    case 'news':
      // checks if "news" service is in the UK or not and set apprioprate destination
      destination = isUK !== false ? 'NEWS_PS' : 'NEWS_GNL';
      break;
    case 'cymrufyw':
    case 'naidheachdan':
      // checks if "cymrufyw" or "naidheachdan" service is in the UK or not and set apprioprate destination
      destination = isUK !== false ? 'NEWS_LANGUAGES_PS' : 'NEWS_LANGUAGES_GNL';
      break;
    case 'japanese':
      destination = 'NEWS_LANGUAGES_GNL';
      break;
    case 'scotland':
      destination = 'PLACEHOLDER';
      break;
    case 'archive':
      destination = 'BBC_ARCHIVE_PS';
      break;
    default:
      destination = 'WS_NEWS_LANGUAGES';
  }
  return env === 'live' ? destination : `${destination}_TEST`;
};

export default getStatsDestination;
