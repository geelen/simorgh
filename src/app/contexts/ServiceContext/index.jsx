import React from 'react';
import { node, string } from 'prop-types';
import services from '#lib/config/services/loadableConfig';
import { getVariant } from '../../lib/utilities/variantHandler';
import variantPropType from '../../models/propTypes/variants';

export const ServiceContext = React.createContext({});

export const ServiceContextProvider = ({ children, service, variant }) => {
  const LoadableContextProvider = services[service];

  if (!LoadableContextProvider) {
    return null;
  }

  return (
    <LoadableContextProvider
      Context={ServiceContext}
      dataKey={getVariant({ service, variant })}
    >
      {children}
    </LoadableContextProvider>
  );
};

ServiceContextProvider.propTypes = {
  children: node.isRequired,
  service: string,
  variant: variantPropType,
};

ServiceContextProvider.defaultProps = {
  service: 'default',
  variant: 'default',
};
