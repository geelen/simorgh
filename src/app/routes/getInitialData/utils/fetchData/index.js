import 'isomorphic-fetch';
import nodeLogger from '#lib/logger.node';
import preprocess from '#lib/utilities/preprocessor';

const logger = nodeLogger(__filename);
const STATUS_CODE_OK = 200;
const STATUS_CODE_BAD_GATEWAY = 502;
const STATUS_CODE_NOT_FOUND = 404;
const upstreamStatusCodesToPropagate = [STATUS_CODE_OK, STATUS_CODE_NOT_FOUND];

const handleResponse = preprocessorRules => async response => {
  const { status } = response;

  return {
    status,
    ...(status === STATUS_CODE_OK && {
      pageData: preprocess(await response.json(), preprocessorRules),
    }),
  };
};

const checkForError = url => ({ status, pageData }) => {
  const isHandledStatus = upstreamStatusCodesToPropagate.includes(status);

  if (isHandledStatus) {
    return { status, pageData };
  }
  logger.warn(
    `Unexpected upstream response (HTTP status code ${status}) when requesting ${url}`,
  );
  throw new Error();
};

const handleError = error => {
  if (error.message) {
    logger.error(error.message);
  }
  return { error, status: STATUS_CODE_BAD_GATEWAY };
};

const fetchData = ({ url, preprocessorRules }) =>
  fetch(url)
    .then(handleResponse(preprocessorRules))
    .then(checkForError(url))
    .catch(handleError);

export default fetchData;
