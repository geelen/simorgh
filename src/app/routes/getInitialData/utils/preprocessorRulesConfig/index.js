import applyTimestampRules from '#lib/utilities/preprocessor/rules/timestamp';
import addIdsToBlocks from '#lib/utilities/preprocessor/rules/addIdsToBlocks';
import radioPageAddIdsToBlocks from '#lib/utilities/preprocessor/rules/radioPage/addIdsToBlocks';
import applyBlockPositioning from '#lib/utilities/preprocessor/rules/blockPositioning';
import parseInternalLinks from '#lib/utilities/preprocessor/rules/cpsAssetPage/convertToOptimoBlocks/blocks/internalLinks';
import convertToOptimoBlocks from '#lib/utilities/preprocessor/rules/cpsAssetPage/convertToOptimoBlocks';
import addHeadlineBlock from '#lib/utilities/preprocessor/rules/cpsAssetPage/addHeadlineBlock';
import timestampToMilliseconds from '#lib/utilities/preprocessor/rules/cpsAssetPage/timestampToMilliseconds';
import filterUnknownContentTypes from '#lib/utilities/preprocessor/rules/filterContentType';
import filterEmptyGroupItems from '#lib/utilities/preprocessor/rules/filterEmptyGroupItems';
import applySquashTopstories from '#lib/utilities/preprocessor/rules/topstories';
import addIdsToItems from '#lib/utilities/preprocessor/rules/addIdsToItems';
import filterGroupsWithoutStraplines from '#lib/utilities/preprocessor/rules/filterGroupsWithoutStraplines';
import addSummaryBlock from '#lib/utilities/preprocessor/rules/cpsAssetPage/addSummaryBlock';

export const cpsAssetPreprocessorRules = [
  parseInternalLinks,
  timestampToMilliseconds,
  convertToOptimoBlocks,
  addHeadlineBlock,
  addSummaryBlock,
  applyTimestampRules,
  addIdsToBlocks,
  applyBlockPositioning,
];

export const articlesPreprocessorRules = [
  applyTimestampRules,
  addIdsToBlocks,
  applyBlockPositioning,
];

export const indexPreprocessorRules = [
  filterUnknownContentTypes,
  filterEmptyGroupItems,
  addIdsToItems,
  applySquashTopstories,
  filterGroupsWithoutStraplines,
];

export const radioPagePreprocessorRules = [radioPageAddIdsToBlocks];
