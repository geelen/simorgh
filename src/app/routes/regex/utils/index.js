const idRegex = 'c[a-zA-Z0-9]{10}o';
const ampRegex = '.amp';
const assetUriRegex = '[a-z-_]{0,}[0-9]{8,}';
const variantRegex = '/simp|/trad|/cyr|/lat';
const articleLocalRegex = 'articles|erthyglau|sgeulachdan';
const mediaIdRegex = '[a-z0-9]+';
const mediaServiceIdRegex = 'bbc_[a-z]+_radio|bbc_[a-z]+_tv';
const errorCodeRegex = '404|500';

const getServiceRegex = services => services.join('|');

export const getArticleRegex = services => {
  const serviceRegex = getServiceRegex(services);
  return `/:service(${serviceRegex})/:local(${articleLocalRegex})/:id(${idRegex}):variant(${variantRegex})?:amp(${ampRegex})?`;
};

export const getArticleSwRegex = services => {
  const serviceRegex = getServiceRegex(services);
  return `/:service(${serviceRegex})/:local(${articleLocalRegex})/sw.js`;
};

export const getArticleManifestRegex = services => {
  const serviceRegex = getServiceRegex(services);
  return `/:service(${serviceRegex})/:local(${articleLocalRegex})/manifest.json`;
};

export const getFrontPageRegex = services => {
  const serviceRegex = getServiceRegex(services);
  return `/:service(${serviceRegex}):variant(${variantRegex})?:amp(${ampRegex})?`;
};

export const getSwRegex = services => {
  const serviceRegex = getServiceRegex(services);
  return `/:service(${serviceRegex})/sw.js`;
};

export const getManifestRegex = services => {
  const serviceRegex = getServiceRegex(services);
  return `/:service(${serviceRegex})/manifest.json`;
};

export const getCpsAssetRegex = services => {
  const serviceRegex = getServiceRegex(services);
  return `/:service(${serviceRegex})/:assetUri(${assetUriRegex}):variant(${variantRegex})?:amp(${ampRegex})?`;
};

export const getRadioAndTVRegex = services => {
  const serviceRegex = getServiceRegex(services);
  return `/:service(${serviceRegex})/:serviceId(${mediaServiceIdRegex})/:mediaId(${mediaIdRegex}):amp(${ampRegex})?`;
};

export const getErrorPageRegex = services => {
  const serviceRegex = getServiceRegex(services);
  return `/:service(${serviceRegex})/:errorCode(${errorCodeRegex}):variant(${variantRegex})?`;
};
