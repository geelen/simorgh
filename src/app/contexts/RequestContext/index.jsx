import React from 'react';
import { bool, node, oneOf, string, number } from 'prop-types';
import getStatsDestination from './getStatsDestination';
import getStatsPageIdentifier from './getStatsPageIdentifier';
import getOriginContext from './getOriginContext';
import getEnv from './getEnv';
import getMetaUrls from './getMetaUrls';
import variantPropType from '../../models/propTypes/variants';

export const RequestContext = React.createContext({});

export const RequestContextProvider = ({
  children,
  bbcOrigin,
  id,
  isAmp,
  pageType,
  service,
  statusCode,
  previousPath,
  pathname,
  variant,
}) => {
  const { isUK, origin } = getOriginContext(bbcOrigin);
  const env = getEnv(origin);
  const platform = isAmp ? 'amp' : 'canonical';
  const statsDestination = getStatsDestination({
    isUK,
    env,
    service,
  });
  const statsPageIdentifier = getStatsPageIdentifier({
    pageType,
    service,
    id,
  });

  const value = {
    env,
    id,
    isUK,
    origin,
    pageType,
    isAmp,
    platform,
    statsDestination,
    statsPageIdentifier,
    statusCode,
    previousPath,
    variant,
    ...getMetaUrls(origin, pathname),
  };

  return (
    <RequestContext.Provider value={value}>{children}</RequestContext.Provider>
  );
};

RequestContextProvider.propTypes = {
  bbcOrigin: string,
  children: node.isRequired,
  id: string,
  isAmp: bool.isRequired,
  pageType: oneOf([
    'article',
    'frontPage',
    'media',
    'error',
    'MAP',
    'FIX',
    'STY',
    'PGL',
  ]).isRequired,
  service: string.isRequired,
  statusCode: number,
  pathname: string.isRequired,
  previousPath: string,
  variant: variantPropType,
};

RequestContextProvider.defaultProps = {
  bbcOrigin: null,
  id: null,
  statusCode: null,
  previousPath: null,
  variant: null,
};
