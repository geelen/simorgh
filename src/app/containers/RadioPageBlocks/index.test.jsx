import React from 'react';
import path from 'ramda/src/path';
import { latin } from '@bbc/gel-foundations/scripts';
import { matchSnapshotAsync } from '@bbc/psammead-test-helpers';
import { BrowserRouter } from 'react-router-dom';
import { ServiceContext } from '#contexts/ServiceContext';
import { RequestContext } from '#contexts/RequestContext';
import RadioPageBlocks from '.';
import amharicPageData from '#data/amharic/bbc_amharic_radio/liveradio';
import preprocessor from '#lib/utilities/preprocessor';
import { radioPagePreprocessorRules } from '#app/routes/getInitialData/utils/preprocessorRulesConfig';

const serviceContextMock = {
  service: 'amharic',
  script: latin,
  lang: 'am',
  translations: { mediaAssetPage: { audioPlayer: 'Audio player' } },
};

describe('Radio Page Blocks', () => {
  it('should match snapshot for Canonical', async () => {
    const pageData = await preprocessor(
      amharicPageData,
      radioPagePreprocessorRules,
    );
    const blocks = path(['content', 'blocks'], pageData);

    await matchSnapshotAsync(
      <ServiceContext.Provider value={serviceContextMock}>
        <RequestContext.Provider
          value={{
            isAmp: false,
            platform: 'canonical',
            pageType: 'media',
            origin: 'http://localhost:7080',
          }}
        >
          <BrowserRouter>
            <RadioPageBlocks blocks={blocks} />
          </BrowserRouter>
        </RequestContext.Provider>
      </ServiceContext.Provider>,
    );
  });

  it('should match snapshot for AMP', async () => {
    const pageData = await preprocessor(
      amharicPageData,
      radioPagePreprocessorRules,
    );
    const blocks = path(['content', 'blocks'], pageData);

    await matchSnapshotAsync(
      <ServiceContext.Provider value={serviceContextMock}>
        <RequestContext.Provider
          value={{
            isAmp: true,
            platform: 'amp',
            pageType: 'media',
            origin: 'http://localhost:7080',
          }}
        >
          <BrowserRouter>
            <RadioPageBlocks blocks={blocks} />
          </BrowserRouter>
        </RequestContext.Provider>
      </ServiceContext.Provider>,
    );
  });
});
