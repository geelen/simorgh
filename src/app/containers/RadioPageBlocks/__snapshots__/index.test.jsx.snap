// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Page Blocks should match snapshot for AMP 1`] = `
.c0 {
  font-size: 1.75rem;
  line-height: 2rem;
  font-family: "Noto Sans Ethiopic",Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 700;
  font-style: normal;
  color: #3F3F42;
  display: block;
  margin: 0;
  padding: 2rem 0;
}

.c4 {
  height: 165px;
  position: relative;
  overflow: hidden;
}

.c3 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 50rem;
  max-width: calc(100vw - 2rem);
}

.c1 {
  font-size: 0.9375rem;
  line-height: 1.25rem;
  font-family: "Noto Sans Ethiopic",Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 400;
  font-style: normal;
  color: #3F3F42;
  padding-bottom: 1.5rem;
  margin: 0;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c0 {
    font-size: 2rem;
    line-height: 2.25rem;
  }
}

@media (min-width:37.5rem) {
  .c0 {
    font-size: 2.75rem;
    line-height: 3rem;
  }
}

@media (min-width:37.5rem) {
  .c0 {
    padding: 2.5rem 0;
  }
}

@media (min-width:63rem) {
  .c2 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c1 {
    font-size: 1rem;
    line-height: 1.375rem;
  }
}

@media (min-width:37.5rem) {
  .c1 {
    font-size: 1rem;
    line-height: 1.375rem;
  }
}

<html>
  <head>
    <script
      async="true"
      custom-element="amp-iframe"
      src="https://cdn.ampproject.org/v0/amp-iframe-0.1.js"
    />
  </head>
  <body>
    <div>
      <h1
        class="c0"
        id="content"
        tabindex="-1"
      >
        ያድምጡ
      </h1>
      <p
        class="c1"
      >
        ዝግጅቶቻችንን’
      </p>
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <div
            class="c4"
          >
            <amp-iframe
              allowfullscreen="allowfullscreen"
              frameborder="0"
              layout="fill"
              sandbox="allow-scripts allow-same-origin"
              scrolling="no"
              src="https://polling.test.bbc.co.uk/ws/av-embeds/media/bbc_amharic_radio/liveradio/am/amp"
              title="Audio player"
            >
              <amp-img
                alt=""
                attribution=""
                height="16"
                layout="fill"
                placeholder="true"
                src="http://localhost:7080/images/amp_audio_placeholder.png"
                width="9"
              />
            </amp-iframe>
          </div>
        </div>
      </div>
    </div>
  </body>
</html>
`;

exports[`Radio Page Blocks should match snapshot for Canonical 1`] = `
.c0 {
  font-size: 1.75rem;
  line-height: 2rem;
  font-family: "Noto Sans Ethiopic",Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 700;
  font-style: normal;
  color: #3F3F42;
  display: block;
  margin: 0;
  padding: 2rem 0;
}

.c4 {
  height: 165px;
  position: relative;
  overflow: hidden;
}

.c3 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 50rem;
  max-width: calc(100vw - 2rem);
}

.c1 {
  font-size: 0.9375rem;
  line-height: 1.25rem;
  font-family: "Noto Sans Ethiopic",Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 400;
  font-style: normal;
  color: #3F3F42;
  padding-bottom: 1.5rem;
  margin: 0;
}

.c5 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  border: 0;
  overflow: hidden;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c0 {
    font-size: 2rem;
    line-height: 2.25rem;
  }
}

@media (min-width:37.5rem) {
  .c0 {
    font-size: 2.75rem;
    line-height: 3rem;
  }
}

@media (min-width:37.5rem) {
  .c0 {
    padding: 2.5rem 0;
  }
}

@media (min-width:63rem) {
  .c2 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c1 {
    font-size: 1rem;
    line-height: 1.375rem;
  }
}

@media (min-width:37.5rem) {
  .c1 {
    font-size: 1rem;
    line-height: 1.375rem;
  }
}

<div>
  <h1
    class="c0"
    id="content"
    tabindex="-1"
  >
    ያድምጡ
  </h1>
  <p
    class="c1"
  >
    ዝግጅቶቻችንን’
  </p>
  <div
    class="c2"
  >
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <iframe
          allow="autoplay; fullscreen"
          allowfullscreen=""
          class="c5"
          scrolling="no"
          src="https://polling.test.bbc.co.uk/ws/av-embeds/media/bbc_amharic_radio/liveradio/am"
          title="Audio player"
        />
      </div>
    </div>
  </div>
</div>
`;
